napari/layers/points/points.py

```diff
@@ -279,18 +405,100 @@ def data(self, data: np.ndarray):
-                new_edge_colors = np.tile(
-                    self._current_edge_color, (adding, 1)
-                )
+
+                for k in self.properties:
+                    new_property = np.repeat(
+                        self.current_properties[k], adding, axis=0
+                    )
+                    self.properties[k] = np.concatenate(
+                        (self.properties[k], new_property), axis=0
+                    )
+
+                # add new edge colors
+                if self._edge_color_mode == ColorMode.DIRECT:
+                    new_edge_colors = np.tile(
+                        self._current_edge_color, (adding, 1)
+                    )
+                elif self._edge_color_mode == ColorMode.CYCLE:
+                    edge_color_property = self.current_properties[
+                        self._edge_color_property
+                    ][0]
+                    new_edge_colors = np.tile(
+                        self.edge_color_cycle_map[edge_color_property],
+                        (adding, 1),
+                    )
+                elif self._edge_color_mode == ColorMode.COLORMAP:
+                    edge_color_property_value = self.current_properties[
+                        self._edge_color_property
+                    ][0]
+
+                    ec, _ = map_property(
+                        prop=edge_color_property_value,
+                        colormap=self.edge_colormap[1],
+                        contrast_limits=self._edge_contrast_limits,
+                    )
+                    new_edge_colors = np.tile(ec, (adding, 1))
                 self.edge_color = np.vstack((self.edge_color, new_edge_colors))
-                new_face_colors = np.tile(
-                    self._current_face_color, (adding, 1)
-                )
+
+                # add new face colors
+                if self._face_color_mode == ColorMode.DIRECT:
+                    new_face_colors = np.tile(
+                        self._current_face_color, (adding, 1)
+                    )
+                elif self._face_color_mode == ColorMode.CYCLE:
+                    face_color_property_value = self.current_properties[
+                        self._face_color_property
+                    ][0]
+                    new_face_colors = np.tile(
+                        self.face_color_cycle_map[face_color_property_value],
+                        (adding, 1),
+                    )
+                elif self._face_color_mode == ColorMode.COLORMAP:
+                    face_color_property_value = self.current_properties[
+                        self._face_color_property
+                    ][0]
+
+                    fc, _ = map_property(
+                        prop=face_color_property_value,
+                        colormap=self.face_colormap[1],
+                        contrast_limits=self._face_contrast_limits,
+                    )
+                    new_face_colors = np.tile(fc, (adding, 1))
```

Comment:
The repetition of edge color and face color code makes me think maybe we could write a function here that uses `getattr` to get the appropriate value. What do you think @kevinyamauchi?